src/tiledb/cloud/utilities/common.py

```diff
@@ -0,0 +1,197 @@
+import configparser
+import logging
+import os
+import sys
+from typing import Any, Mapping
+
+import tiledb
+import tiledb.cloud
+
+
+def read_aws_config(
+    path: str = "~/.aws/credentials",
+    section: str = "default",
+) -> Mapping[str, Any]:
+    """
+    Read config values from a file.
+
+    :param path: config file, defaults to "~/.aws/credentials"
+    :param section: section to read in the config file, defaults to "default"
+    :return: config dictionary
+    """
+    config = {}
+
+    cp = configparser.ConfigParser()
+    cp.read(os.path.expanduser(path))
+
+    keys = [
+        "aws_access_key_id",
+        "aws_secret_access_key",
+        "aws_session_token",
+        "aws_role_arn",
+        "aws_role_session_name",
+        "aws_external_id",
+    ]
+
+    for key in keys:
+        try:
+            config["vfs.s3." + key] = cp[section][key]
+        except KeyError:
+            pass
+
+    return config
+
+
+def set_aws_context(config: Mapping[str, Any] = {}):
```

Comment:
While it looks like `config` is never mutated, I'm still uneasy with mutable objects as defaults. Maybe just `None`?

Also, add `-> None` to these functions which return nothing.